### PR TITLE
4892- Add committee_type filter and test

### LINKED ISSUE #4892: Rename committee type filter in `/reports/entity_type/` from `type` to `committee_type`
### What we’re after

To be consistent across endpoints, the `committee_type` filter would ideally be named `committee_type` everywhere. However, the committee type filter in `/reports/entity_type/` is called `type`. This is confusing to the users and can be a source of tech debt when things aren't consistent.

We should notify API users and rename committee type filter in `/reports/entity_type/` from `type` to `committee_type`

### Related ticket(s)
https://github.com/fecgov/openFEC/issues/4886

### Action item(s)

(These are the smaller tasks that should happen in order to complete this work)
- [x] Add new `committee_type` filter
- [x] Let API users that have used this endpoint know to use the new filter (follow deprecation policy here: https://github.com/fecgov/openFEC/wiki/Deprecation-policy)
- [x] Check what change need to be made in the CMS. Update it to the new name before the old name is removed
- [x] Create ticket to remove old `type` filter and add validation errors

### Completion criteria

- [x] Committee `type` filter in `/reports/entity_type/` is named `committee_type`

### PR DESCRIPTION
**Pull request by cnlucas**
## Summary (required)

- Resolves #4892 

Adds the `committee_type` filter to gradually change `type` to `committee_type` per our deprecation policy. Also, updates some test names to match with the `entity_type` change. Also, adds a test for the new filter (thanks @pkfec!)

### Required reviewers

1-2 devs

## Impacted areas of the application

General components of the application that this PR will affect:

-  financials/reports/{entity_type} endpoint 

## Screenshots
Screen Shot 2022-02-03 at 2 53 46 PM

## How to test

- Check out this branch
- Try queries for financials/reports/<string:entity_type>/    using the old `type` and new `committee_type` filters, should work with both, either, or neither!
- http://localhost:5000/v1/reports/pac-party/?sort_nulls_last=false&sort=-committee_id&sort_null_only=false&per_page=20&sort_hide_null=false&api_key=DEMO_KEY&cycle=2022&page=1

### CODE REVIEW

**Review by pkfec — COMMENTED** [review]
`type` filter  is replaced with `committee_type` filter. `type` filter is no longer needed for this endpoint. You can remove this line.

File: webservices/args.py [hunk 1]
```diff
@@ -377,6 +377,7 @@ def make_seek_args(field=fields.Int, description=None):
     'min_total_contributions': Currency(description=docs.MIN_FILTER),
     'max_total_contributions': Currency(description=docs.MAX_FILTER),
     'type': fields.List(fields.Str, description=docs.COMMITTEE_TYPE),
```
Comment (pkfec) [inline webservices/args.py:379]: `type` filter is replaced by `committee_type` filter. `type` filter is no longer needed for this endpoint. You can remove this line.

File: webservices/resources/reports.py [hunk 2]
```diff
@@ -181,7 +181,10 @@ def build_query(self, entity_type=None, **kwargs):
             query = query.filter(
```
Comment (pkfec) [inline webservices/resources/reports.py:181]: same here. Remove `type` filter logic

File: webservices/args.py (hunk 1 above)
Comment (cnlucas) [reply webservices/args.py:379]: Will be done after users are notified due to breaking change as per our deprecation policy and my convo with @pkfec

File: webservices/resources/reports.py (hunk 2 above)
Comment (cnlucas) [reply webservices/resources/reports.py:181]: ^will be done after users are notified due to breaking change as per our deprecation policy

**Comment by cnlucas** [comment]
> `type` filter is replaced with `committee_type` filter. `type` filter is no longer needed for this endpoint. You can remove this line.

Users will be notified before 2/22 release and per our deprecation policy both filters will be usable for at least 30 days. Ticket to remove the logic has been created. https://github.com/fecgov/openFEC/issues/5043

**Comment by fec-jli** [comment]
Good job.
We should put `deprecated` (pink color) next the `type` filter or in type description section. after 30 days, remove type filter.

**Comment by cnlucas** [comment]
> Good job. @cnlucas We should put `deprecated` (pink color) next the `type` filter or in type description section. after 30 days, remove type filter.

Thank you, added!

**Review by fec-jli — APPROVED** [review]
good job. Thanks

**Review by hcaofec — APPROVED** [review]
Looks great! Approved.

**Comment by cnlucas** [comment]
Thanks @hcaofec, @pkfec  and @fec-jli!

**Pull request merged**